pkg/kubectl/cmd/edit.go

```diff
@@ -354,7 +350,9 @@ func getPrinter(cmd *cobra.Command) (*editPrinterOptions, error) {
 	}
 }
 
-func getMapperAndResult(f cmdutil.Factory, args []string, options *resource.FilenameOptions, editMode EditMode) (meta.RESTMapper, *resource.Mapper, *resource.Result, string, error) {
+type resultGetter func([]byte) *resource.Result
+
+func getMapperAndResult(f cmdutil.Factory, args []string, options *resource.FilenameOptions, editMode EditMode) (meta.RESTMapper, *resource.Result, resultGetter, string, error) {
```

Comment:
Returning 5 results makes this hard to read.  WDYT of returning a struct + error?  The struct then could have resultGetter as a function on the struct itself.  Later we might be able to refactor other functions into the struct.